spec/features/workflow/unpublishing_spec.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+
+require "rake"
+
+RSpec.feature "Unpublish documents rake tasks" do
+  describe "unpublish:retire_document" do
+    before do
+      Rake::Task["unpublish:retire_document"].reenable
+    end
+
+    it "runs the task to retire a document" do
+      document = create(:document, :published, locale: "en")
+      explanatory_note = "The reason the document is being retired"
+
+      expect_any_instance_of(DocumentUnpublishingService).to receive(:retire).with(document, explanatory_note, locale: "en")
+
+      ClimateControl.modify CONTENT_ID: document.content_id, NOTE: explanatory_note do
+        Rake::Task["unpublish:retire_document"].invoke
+      end
+    end
+
+    it "raises an error if a CONTENT_ID is not present" do
+      expect { Rake::Task["unpublish:retire_document"].invoke }.to raise_error("Missing CONTENT_ID value")
+    end
+
+    it "raises an error if a NOTE is not present" do
+      ClimateControl.modify CONTENT_ID: "a-content-id" do
+        expect { Rake::Task["unpublish:retire_document"].invoke }.to raise_error("Missing NOTE value")
+      end
+    end
+
+    it "copes with commas in the explanatory note" do
```

Comment:
I'm super pleased you checked for this but I don't think we actually need a test in the code for it.

Mainly as it's not realistic as it's down to bash's string handling rather than in the ruby.

Plus since we're using an env var and not multiple rake arguments we're free from that concern